restarter/hot-restarter.py

```diff
@@ -5,18 +5,69 @@
 import sys
 import time
 
+# The number of seconds to wait for children to gracefully exit after
+# propagating SIGTERM before force killing children.
+# NOTE: If using a shutdown mechanism such as runit's `force-stop` which sends
+# a KILL after a specified timeout period, it's important to ensure that this
+# constant is smaller than the KILL timeout
+TERM_WAIT_SECONDS = 30
```

Comment:
should I make this configurable via a flag? My assumption was that that overly complicates this script, since it's likely it will be customized anyway

Reply:
I think this is fine, we can modify this later if we need more configurability.